runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -110,6 +110,14 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
   return vim.tbl_map(function(diagnostic)
     local start = diagnostic.range.start
     local _end = diagnostic.range['end']
+    for _, relatedInfo in pairs(diagnostic.relatedInformation or {}) do
+      if relatedInfo.location then
```

Comment:
`location` seems to be a non-nil (required) field of the `lsp.DiagnosticRelatedInformation` class.